src/parser/ParsedQuery.h

```diff
@@ -138,6 +138,39 @@ class VariableOrderKey {
 // Represents an ordering by a variable or an expression.
 using OrderKey = std::variant<VariableOrderKey, ExpressionOrderKey>;
 
+/// Store an expression that appeared in a GROUP BY clause.
+class ExpressionGroupKey {
+ public:
+  sparqlExpression::SparqlExpressionPimpl expression_;
+  // ___________________________________________________________________________
+  explicit ExpressionGroupKey(
+      sparqlExpression::SparqlExpressionPimpl expression)
+      : expression_{std::move(expression)} {}
+};
```

Comment:
For these simple aggregates (all members are public) you can simply write
```
// struct implies public
struct ExpressionGroupKey {
  SparqlExpressionPimpl expression_;
}
```

You can then initialize them using braces `{}`:
```
ExpressionGroupKey k{std::move(somePimpl)};
```
(The parentheses `ExpressionGroupKey k(...)` won't work until C++23, because you don't have a constructor but use "aggregate initialization"